internal/handlers/commonparams/extract_params.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commonparams
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"go.uber.org/zap"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/iterator"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// ExtractParams fill passed value structure with parameters from the document.
+// If the passed value is not a pointer to the structure it panics.
+// Parameters are extracted by the field name or by the `name` tag.
+//
+// Possible tags:
+//   - `opt` - field is optional;
+//   - `non-default` - field is unimplemented non-default;
```

Comment:
This description needs improving. Field is unimplemented for non-default values? :thinking:

Reply:
Fixed